File: 김주영은보아라.swift
class Solution {
    func rotate(_ nums: inout [Int], _ k: Int) {
        var temp = 0
        var temp2 = nums[0]
        let len = nums.count
        var idx = 0
        var cnt = 0
        for i in 0..<len {
            let val = (idx + k) % len
            temp = nums[val]
            nums[val] = temp2
            temp2 = temp
            idx = val
            if idx == cnt, len > 1 {
                if cnt == nums.count - 1 {
                    continue
                }
                cnt += 1
                idx = cnt
                temp = 0
                temp2 = nums[idx]
            }
        }
        
    }
}

File: 김주영은보아라.cpp
#include<iostream>
#include <vector>
using namespace std;

int main() {
	vector<int> v = { 0,1,2,3,4,5,6 };
	int m;
	cin >> m;
	int len = v.size();
	int temp = 0;
	
	int idx = 0;
	int temp2 = v[idx];
	for (int i = 0; i < len; i++) {
		int val = (idx + m) % len;
		temp = v[val];
		v[val] = temp2;
		temp2 = temp;
		idx = val;
	}
	for (int i = 0; i < len; i++) {
		cout << v[i] << ' ';
	}cout << '\n';
}
